From: =?utf-8?q?Timm_B=C3=A4der?= <mail@baedert.org>
Date: Sat, 3 Jun 2017 14:38:56 +0200
Subject: popover: Remove useless if case

rect is not nullable, so stop checking if it is NULL and instead add a
g_return_val_if_fail guard.

Origin: 3.24.39, commit:ad07e9043f36319ae2d733a8d5d0ef89ee9a7586
---
 gtk/gtkpopover.c | 14 ++++++--------
 1 file changed, 6 insertions(+), 8 deletions(-)

diff --git a/gtk/gtkpopover.c b/gtk/gtkpopover.c
index 45680d0..82ad2fc 100644
--- a/gtk/gtkpopover.c
+++ b/gtk/gtkpopover.c
@@ -2327,16 +2327,14 @@ gtk_popover_get_pointing_to (GtkPopover   *popover,
   GtkPopoverPrivate *priv = gtk_popover_get_instance_private (popover);
 
   g_return_val_if_fail (GTK_IS_POPOVER (popover), FALSE);
+  g_return_val_if_fail (rect != NULL, FALSE);
 
-  if (rect)
+  if (priv->has_pointing_to)
+    *rect = priv->pointing_to;
+  else if (priv->widget)
     {
-      if (priv->has_pointing_to)
-        *rect = priv->pointing_to;
-      else if (priv->widget)
-        {
-          gtk_widget_get_allocation (priv->widget, rect);
-          rect->x = rect->y = 0;
-        }
+      gtk_widget_get_allocation (priv->widget, rect);
+      rect->x = rect->y = 0;
     }
 
   return priv->has_pointing_to;
